_ramblings/2019-01-05-splitting-cats-and-dogs.md

```diff
@@ -0,0 +1,326 @@
+---
+title: Splitting Cats and Dogs
+layout: rambling
+draft: true
+---
+
+Often when you're programming you find the need to split things up a bit. This
+is such a fundamental part of programming that almost every language gives you
+many tools to aid with this. Maybe it's _modules_, _classes_, _impls_, or
+even _packages_. The list goes on... In fact, one of the best forms of
+splitting can be a simple _function_. Even _comments_ can play a role in this
+if you'd like.
+
+There's a number of reasons you'd want to split up some part of a program. You
+might be looking at a class with the need for countless functions. Instead of
+defining them all one after another, it makes sense to _group_ the functions
+and move them somewhere so it's clear how they are related. Or, you might be
+looking to _share_ some logic across a few classes. This is key to writing
+good software, since maintaining two versions of the same thing is cumbersome.
+Remember, stay DRY (Don't Repeat Yourself).
+
+### Grouping
+
+Grouping shouldn't require too much thought, aside from a **name** for the way
```

Comment:
the argument here is an over simplification, this here only implied cohesion, but you also need to think about decoupling and dependency management, try to see if this concepts and be alluded briefly somehow.

Reply:
But do we really need to bring all these terms into play, I'm leaving a lot implict because frankly it's not critical. How does dependency management play into this here, it's very subtle. The only management required is done from within Ruby's internals, and most languages do the same kinds of things. The fact that namespace resolution works is neat, and a fun topic in it of itself, but frankly outside the scope of this.

I've tweaked the language of this sentence to mention cohesion off-hand, as I do feel this is a good descriptor.